controller/filehandlingcontroller.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * @author Björn Schießle <schiessle@owncloud.com>
+ *
+ * @copyright Copyright (c) 2015, ownCloud, Inc.
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+
+namespace OCA\Files_Texteditor\Controller;
+
+
+use OC\Files\View;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\IL10N;
+use OCP\ILogger;
+use OCP\IRequest;
+
+class FileHandlingController extends Controller{
+
+	/** @var IL10N */
+	private $l;
+
+	/** @var View */
+	private $view;
+
+	/** @var ILogger */
+	private $logger;
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @param string $AppName
+	 * @param IRequest $request
+	 * @param IL10N $l10n
+	 * @param View $view
+	 * @param ILogger $logger
+	 */
+	public function __construct($AppName,
+								IRequest $request,
+								IL10N $l10n,
+								View $view,
+								ILogger $logger) {
+		parent::__construct($AppName, $request);
+		$this->l = $l10n;
+		$this->view = $view;
+		$this->logger = $logger;
+	}
+
+	/**
+	 * load text file
+	 *
+	 * @NoAdminRequired
+	 *
+	 * @param string $dir
+	 * @param string $filename
+	 * @return DataResponse
+	 */
+	public function load($dir, $filename) {
+		try {
+			if (!empty($filename)) {
+				$path = $dir . '/' . $filename;
+				$writable = $this->view->isUpdatable($path);
+				$mime = $this->view->getMimeType($path);
+				$mtime = $this->view->filemtime($path);
+				$filecontents = $this->view->file_get_contents($path);
+				if ($filecontents) {
```

Comment:
Now you can not edit files anymore that are empty, or `0`

Suggestion, replace the code until here with:
```php
		try {
			if (!empty($filename)) {
				$path = $dir . '/' . $filename;
				if ($this->view->file_exists($path)) {
					$writable = $this->view->isUpdatable($path);
					$mime = $this->view->getMimeType($path);
					$mtime = $this->view->filemtime($path);
					$filecontents = $this->view->file_get_contents($path);
```